q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,322 @@
+        Path to one variable's rules yaml file.
+
+    Attributes
+    ----------
+    variable : str
+        Name of the variable which rules are parsed and
+        that is the yaml file name without the .yml extension.
+
+    remap : dict
+        Apply the rules of remapping variables values, by replacing
+        all instances of a given value by another one value.
+                    'remap:
+                      str: str'
+    validation : dict
+        Apply the rules of validating variables values, by replacing by the blank
+        (or the missing) value every instance which in one or many other columns
+        have a null or missing value.
+                    'validation:
+                      force_to_blank_if:
+                        is null:
+                        - variable'
+    normalization : dict
+        Apply the rules of normalizing the variables values, by replacing
+        the values outside the range by a given replacement value.
+                    'normalization:
+                        gated_value:
+                            - Out of bounds
+                        maximum:
+                            - int
+                            - float
+                        minimum:
+                            - int
+                            - float'
+    missing : str
+        Apply the rules of checking the allowed missing values, by replacing
+        the not-expected values and numpy's nan (empty) by the passed
+        missing value term.
+                    'missing:
+                        - not applicable
+                        - not collected
+                        - not provided
+                        - restricted access'
+    format : str
+        Apply the rules of checking that the passed column in the correct format.
+        [THIS COMMAND WOULD EITHER INFER THE DTYPE (as of now) OR USE THE
+        RESULT OF AN INITIAL DTYPE GETTER FUNCTION (see _dtypes.py)]
+                    'format:
+                      - bool
+                      - float
+                      - int
+                      - str'
+    expected : list
+        Apply the rules of allowing only expected values.
+        These do allow the missing and blanks.
+                    'expected:
+                        - str
+                        - list'
+    ontology : str
+        Name of the ontology to lookup within for control vocabulary, e.g.
+                    'ontology: Gazetteer'
+    blank : str
+        Apply the rules of checking the allowed blank values, by replacing
+        the not-expected values and numpy's nan (empty) by the passed
+        blank value term, e.g.
+                    'blank:
+                        - not applicable
+                        - not collected
+                        - not provided
+                        - restricted access'
+
+    Raises
+    ------
+    ValueError
+        # If the remote argument is not of ``bool`` or ``str`` type.
```

Comment:
what is a remote argument?

Reply:
replaced by the actual raised errors (that was the "remains" from a copy-paste from another package... sorry)
see in:
https://github.com/ElDeveloper/q2-metadata/pull/3/commits/e2f893e510b6a1baf5fb9e25cd9c2d721c87c5e9
(line 242 in the "green" version)

```
    Raises
    ------
    IOError
        If directory passed to --p-rules-dir does not exist.
    FileNotFoundError
        If directory passed to --p-rules-dir contains no .yml file.
```